generators/server/templates/_pom.xml

```diff
@@ -616,6 +616,16 @@
             <artifactId>spring-retry</artifactId>
         </dependency>
         <%_ } _%>
+        <%_ if (applicationType === 'microservice' && authenticationType === 'uaa') { _%>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-security</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-feign</artifactId>
```

Comment:
I could carry out the feign block to general (since it might be interesting for monoliths also)

but cloud security is really needed when working with uaa